File: src/easy/242-valid-anagram/validAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        lenS = len(s)

        if (lenS != len(t)):
          return False

        list = [0]*26

        for i in range(lenS):
          list[ord(s[i]) - 97] += 1
          list[ord(t[i]) - 97] -= 1
        
        for x in list:
          if x > 0:
            return False
        
        return True

testSolution = Solution()

print(testSolution.validAnagram("anagram", "nagaram")); # output == true
print(testSolution.validAnagram("rat", "car"));         # output == false

File: src/easy/242-valid-anagram/validAnagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
  if (s.length !== t.length) {
    return false;
  }

  const counts = new Array(26).fill(0);

  for (let i = 0; i < s.length; i++) {
    counts[s[i].charCodeAt(0) - 97]++;
    counts[t[i].charCodeAt(0) - 97]--;
  }

  for (let i = 0; i < counts.length; i++) {
    if (counts[i] > 0) {
      return false;
    }
  }

  return true;
};

console.log(isAnagram("anagram", "nagaram")); // output == true
console.log(isAnagram("rat", "car"));         // output == false